catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -355,9 +358,42 @@ private String getExceptionMessage(String responseType, String previousException
         return builder.toString();
     }
 
+    public void shutdown() {
+        getPeriodicDrain().shutdown();
+    }
+
+    public String getRootBackupDir() {
+        return rootBackupDir.getAbsolutePath();
+    }
+
+    /**
+     * Sets the root file system backup directory.
+     *
+     * @param dir absolute path for the root file system backup directory.
+     */
+    public void setRootBackupDir(String dir) {
+        if (StringUtils.isBlank(dir)) {
+            LOGGER.error("The root backup directory is blank.");
```

Comment:
Just saw your reply @ahoffer. The backup dir is a container-managed property that gets set when the admin changes things in the UI. We want to provide feedback as soon as the value gets set, otherwise things fail at the wrong time, i.e., when the plug-in is called, and then nobody knows things are misconfigured unless they look at the logs. In other words, I would argue that the unit tests are wrong and should be changed :-)